tests/unit/test_utils.py

```diff
@@ -408,17 +408,63 @@ def test_match_field_dl_vlan(self):
         assert not result
         result = utils.match_field_dl_vlan(None, "4096/4096")
         assert not result
-        result = utils.match_field_dl_vlan(10, 0)
+        result = utils.match_field_dl_vlan([10], 0)
         assert not result
-        result = utils.match_field_dl_vlan(10, 10)
+        result = utils.match_field_dl_vlan([10], 10)
         assert result
-        result = utils.match_field_dl_vlan(10, "4096/4096")
+        result = utils.match_field_dl_vlan([10], "4096/4096")
         assert result
-        result = utils.match_field_dl_vlan(10, 11)
+        result = utils.match_field_dl_vlan([10], 11)
         assert not result
-        result = utils.match_field_dl_vlan(3, "5/1")
+        result = utils.match_field_dl_vlan([3], "5/1")
         assert result
-        result = utils.match_field_dl_vlan(2, "5/1")
+        result = utils.match_field_dl_vlan([2], "5/1")
+        assert not result
+
+    def test_match_field_ip(self):
+        """Test match_field_ip"""
+        # IPv4 cases
+        result = utils.match_field_ip('192.168.20.21', '192.168.20.21')
+        assert result
+        result = utils.match_field_ip('192.168.20.21/10', '192.168.20.21/10')
+        assert result
+        result = utils.match_field_ip('192.168.20.21', '192.168.20.21/32')
+        assert result
+        result = utils.match_field_ip(
+                                        '192.168.20.21',
+                                        '192.168.20.21/255.255.255.255'
+                                    )
+        assert result
+        result = utils.match_field_ip('192.168.20.30', '192.168.20.21')
+        assert not result
+        result = utils.match_field_ip('192.168.20.21', '192.168.20.21/10')
+        assert not result
```

Comment:
This assert is wrong, `'192.168.20.21'` belongs to `'192.168.20.21/10'`, `match_field_ip` isn't correctly implemented for this case. Similarly to `dl_vlan` mask logical and, `packet_ip & flow_ip.hostmask` == `flow_ip.network_address`. If logical and hasn't been implemented for the stdlib you need to derive it.

Reply:
Fine, I assumed mask /32 for the case where it is not specified, so `'192.168.20.21' == '192.168.20.21/32'`. Anyway, I am going to change the logic according to your comment.

Reply:
@gretelliz, mask /32 is only fine for flow_field when it doesn't have any, the input shouldn't not have a mask since the input is a packet it should always be a valid IP host, technically the input will always be /32, we shouldn't not support a mask (or range) in the input, same reasoning as we don't allowed a range of vlans in the input.